image_view/src/nodes/image_view.cpp

```diff
@@ -42,13 +97,40 @@ int main(int argc, char **argv)
              "\t$ rosrun image_view image_view image:=<image topic> [transport]");
   }
 
-  nodelet::Loader manager(false);
-  nodelet::M_string remappings;
-  nodelet::V_string my_argv(argv + 1, argv + argc);
-  my_argv.push_back("--shutdown-on-close"); // Internal
+  ros::NodeHandle nh;
+  ros::NodeHandle local_nh("~");
 
-  manager.load(ros::this_node::getName(), "image_view/image", remappings, my_argv);
+  // Default window name is the resolved topic name
+  std::string topic = nh.resolveName("image");
+  std::string window_name;
+  local_nh.param("window_name", window_name, topic);
+
+  std::string format_string;
+  local_nh.param("filename_format", format_string, std::string("frame%04i.jpg"));
+  g_filename_format.parse(format_string);
+
+  bool autosize;
+  local_nh.param("autosize", autosize, false);
+  cv::namedWindow(window_name, autosize ? cv::WND_PROP_AUTOSIZE : 0);
```

Comment:
also use keep_ratio and maybe gui_expanded ? http://docs.opencv.org/2.4/modules/highgui/doc/user_interface.html#namedwindow